src/recursive_sum/__init__.py

```diff
@@ -0,0 +1,6 @@
+def recursive_sum(arr):
+    if not arr:
+        return 0
+    if len(arr) == 1:
```

Comment:
it is unnecessary, previous case is sufficient

Reply:
No Man. Previous case checks for a list emptiness

Reply:
If this case will be absent and arr's len will be 1, next case happens. Let reduce it: `return arr[0] + recursive_sum(arr[1:])` => `return arr[0] + recursive_sum([])` => `return arr[0] + 0` => `return arr[0]`. Bingo! They are identical
You can investigate various FP libraries and you will see coders excludes superfluous cases can be infered from other cases

Reply:
Ah! I got it

Reply:
Sorry, man, my first comment was inaccurate, it confused you